### PR TITLE
feat: camunda app connection and secret sync

### PR DESCRIPTION
**Pull request by sheensantoscapadngan**
# Description 📣
This PR adds app connection and secret sync for Camunda

## Type ✨

- [ ] Bug fix
- [x] New feature
- [ ] Improvement
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

## Summary by CodeRabbit

- **New Features**
  - Integrated Camunda as a new connection option, enabling users to create, update, and manage Camunda connections using client credentials.
  - Extended secret synchronization functionality to support Camunda clusters, including selecting and syncing secrets with the appropriate cluster.
  - Introduced new components for handling Camunda-specific forms and displaying synchronization details.
  - Added a new component for managing Camunda connection forms and a dedicated section for Camunda sync destinations.

- **Documentation**
  - Added comprehensive API reference pages and integration guides for Camunda, covering connection endpoints and secret synchronization operations.

### CODE REVIEW

File: backend/src/services/secret-sync/camunda/camunda-sync-fns.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+    for await (const secret of Object.keys(camundaSecrets)) {
+      if (!(secret in secretMap)) {
+        await deleteCamundaSecret({
+          key: secret,
+          clusterUUID,
+          accessToken
+        });
+      }
+    }
```
[the remark this answers is not in the document]
Comment (sheensantoscapadngan) [reply backend/src/services/secret-sync/camunda/camunda-sync-fns.ts:121]: copy

File: backend/src/services/app-connection/camunda/camunda-connection-schema.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  BaseAppConnectionSchema,
+  GenericCreateAppConnectionFieldsSchema,
+  GenericUpdateAppConnectionFieldsSchema
+} from "@app/services/app-connection/app-connection-schemas";
+
+import { CamundaConnectionMethod } from "./camunda-connection-enums";
+
+const BaseCamundaConnectionSchema = BaseAppConnectionSchema.extend({ app: z.literal(AppConnection.Camunda) });
+
+export const CamundaConnectionClientCredentialsInputCredentialsSchema = z.object({
```
Comment (carlosmonastyrski) [inline backend/src/services/app-connection/camunda/camunda-connection-schema.ts:15]: We are now also documenting these fields on .describe(AppConnections.CREDENTIALS.Name_Of_Connection.field)

File: backend/src/services/secret-sync/camunda/camunda-sync-fns.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            clusterUUID,
+            accessToken
+          });
+        }
+      } catch (error) {
+        throw new SecretSyncError({
+          error,
+          secretKey: key
+        });
+      }
```
Comment (carlosmonastyrski) [inline backend/src/services/secret-sync/camunda/camunda-sync-fns.ts:108]: Small comment on this part of the logic, it seems like Camunda is expecting a non empty string on the secretValue, maybe we can either omit those from the sync operation or at least send a `" "` value instead. On both approaches I think we should show this on the UI and Docs.

CleanShot 2025-04-10 at 16 11 43

File: docs/api-reference/endpoints/app-connections/camunda/get-by-name.mdx [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (carlosmonastyrski) [inline docs/api-reference/endpoints/app-connections/camunda/get-by-name.mdx:1]: Remember to add these`api-reference/endpoints/app-connections/` on the mint.json file

File: frontend/src/pages/organization/AppConnections/AppConnectionsPage/components/AppConnectionForm/CamundaConnectionForm.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+          name="credentials.clientId"
+          control={control}
+          shouldUnregister
+          render={({ field, fieldState: { error } }) => (
+            <FormControl
+              errorText={error?.message}
+              isError={Boolean(error?.message)}
+              label="Client ID"
+            >
+              <Input {...field} placeholder="05810c8f-c44d-4bd0-a327-aab6dd77719b" />
```
Comment (carlosmonastyrski) [inline frontend/src/pages/organization/AppConnections/AppConnectionsPage/components/AppConnectionForm/CamundaConnectionForm.tsx:106]: Maybe we can shape the placeholder to something that looks a little bit more to the actual clientId generated by Camunda? Like "F_DE344Fhb332" (you can omit this, I just felt it a little bit weird while using it)
CleanShot 2025-04-10 at 16 02 57

File: frontend/src/pages/secret-manager/IntegrationsListPage/components/SecretSyncsTab/SecretSyncTable/helpers/index.ts [hunk 6]
```diff
@@ -73,6 +73,10 @@ export const getSecretSyncDestinationColValues = (secretSync: TSecretSync) => {
       }
       secondaryText = `Organization - ${destinationConfig.org}`;
       break;
+    case SecretSync.Camunda:
+      primaryText = destinationConfig.clusterUUID;
```
Comment (carlosmonastyrski) [inline frontend/src/pages/secret-manager/IntegrationsListPage/components/SecretSyncsTab/SecretSyncTable/helpers/index.ts:77]: Probably would be better for users to see the cluster name instead

**Comment by carlosmonastyrski** [comment]
Also one thing I found while testing it locally is that the API endpoints seems to be empty, maybe it's just my local config but wanted to comment it.

CleanShot 2025-04-10 at 16 53 56

**Review by carlosmonastyrski — CHANGES_REQUESTED** [review]
Mostly nits, a few concerns

**Comment by carlosmonastyrski** [comment]
Everything looks great, just a minor comment coderabbit made about a rename for a component, but LGTM 👍

**Review by carlosmonastyrski — APPROVED** [review]
